asm/alu.go

```diff
@@ -39,46 +39,60 @@ const (
 
 // ALUOp are ALU / ALU64 operations
 //
-//	msb      lsb
-//	+----+-+---+
-//	|OP  |s|cls|
-//	+----+-+---+
-type ALUOp uint8
+//	msb              lsb
+//	+-------+----+-+---+
+//	|  EXT  | OP |s|cls|
+//	+-------+----+-+---+
+type ALUOp uint16
 
-const aluMask OpCode = 0xf0
+const (
+	aluMask    OpCode = 0xfff0
+	aluOpMask  ALUOp  = 0x00f0
```

Comment:
Why not just extend `aluOpMask` to be `0x01f0`? Then the setters, getters should work out of the box and `aluExtMask` isn't necessay anymore.

Reply:
Because this also works with the sign extended mov which has 0x08, 0x10 and 0x20 in its upper bits. The semantic being that the upper byte of the ALUOp is the lower byte of the Offset. That makes going from one to the other easier.